File: code_eval/java/moderate/Challenge27.java
package moderate;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 27 : Decimal to Binary
 * Created by cschalk on 3/21/16.
 */
public class Challenge27 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            decimalToBinary(line);
        }
    }

    private static void decimalToBinary(String input) {
        String binary = Integer.toBinaryString(Integer.parseInt(input));
        System.out.println(binary);
    }
}

File: code_eval/cpp/moderate/challenge_27.cpp
// Created by cschalk on 3/30/16.
// Challenge 27: Decimal to Binary
#include <iostream>
#include <fstream>
#include <bitset>

int main(int argc, char ** argv) {
    std::ifstream input {argv[1]};
    int test;

    while (input >> test) {
        if (test > 0) {
            std::string binary = std::bitset<64>(test).to_string();
            int index = binary.find_first_of('1');
            std::cout << binary.substr(index, binary.length() - index) << std::endl;
        } else {
            std::cout << 0 << std::endl;
        }
    }
    return 0;
}
